### ISSUE TITLE
Adding Sirius Joyport support

### ISSUE THREAD

**Issue by cskare** [AppleWin/AppleWin#1102]
I noticed the Sirius Joyport is coded into AppleWin, but it looks like it isn't enabled.  Can this be added?  A lot of games offer this as a joystick option.

**Comment by tomcw** [comment]
Are there any games that support Joyport, but not the analog joystick?

I'm just wondering: what's the benefit for supporting the Joyport? ...as under emulation either USB digital controllers or the keyboard's cursor keys can be used as digital inputs to mimic min/max analog inputs. So (from within the game) just pick "Apple Joystick", and use one of these inputs to get the same experience as with a Joyport.

And on a real Apple //e, the Joyport will cause the machine to self-test on a RESET - it was only really designed with the Apple II and II+ in mind. [*]

---

[*] http://lukazi.blogspot.com/2016/09/apple-ii-4play-joystick-card-joyport.html

>The Joyport works well on the Apple II and II+ machines but not on the IIe and IIGS which accounts for the majority of today's Apple II users (the IIc doesn't count because it can't use the Joyport ie it doesn't have the 16 pin Game Port socket). When attached to a IIe or IIGS the Joyport causes the machine to go into self test mode on reboot. This happens because the self test mode is tied to the Open and Close Apple keys / paddle buttons and the Joyport sets these paddle buttons to high by default.

**Comment by cskare** [comment]
I'm not aware of games that only support the Joyport.  There's a good write up of the Joyport and the games that support it at:

[ http://nerdlypleasures.blogspot.com/2021/08/digital-joysticks-and-apple-ii.html](url)

Some games work better with the Joyport such as Bandits.  When using an Apple joystick, you still have to hit the Spacebar to enable the shield.  With the Joyport, you just push up on the joystick to enable the shield.  Lastly, with the Apple joystick, it is very difficult to stop your ship at an exact position.  With the Joyport, the ship stops as soon as you release the joystick.

The MAME emulator offers the -gameio joyport option and it works great!  I'd love to see AppleWin offer the same support.

**Comment by cskare** [comment]
Yes, from my experience MAME only supports the Joyport for the Apple II and II+, not the //e,  IIc or IIGS .

For the //e there is the risk on the self-testing infinite loop.  I'm not sure if you can code around the issue in an emulator.

If you do a search of the AppleWin code base for the word "Joyport", you'll see there is already code written for it, so it might not be too much effort to implement it.
